### ISSUE TITLE
Does not render some UTF8 characters

### ISSUE THREAD

**Issue by gullevek** [jwilm/alacritty#1100]
Which operating system does the issue occur on?

macOS 10.12.6

sample line from Terminal.app visible:
set listchars=tab:\→\ ,trail:▫︎,nbsp:•,eol:¶,extends:»,precedes:«

in Alcritty
listchars=tab:\�F~R\ ,trail:�V�N,nbsp:�@,eol:¶,extends:»,precedes:«

**Comment by chrisduerr** [comment]
Does #1099 fix this?

**Comment by gullevek** [comment]
I will try.

But how do I clean up the whole build try to build again? make clean seems to only remove the build app.

**Comment by jwilm** [comment]
@gullevek you can `cargo clean` but that shouldn't be necessary.

**Comment by sodiumjoe** [comment]
it's merged, so you can just pull master

**Comment by sodiumjoe** [comment]
@gullevek were you able to rebuild from latest master?

**Comment by gullevek** [comment]
I pulled in the updates and rebuild (in just make app and also with cargo clean) but I still cannot input japanese nor view japanese characters.

Pressing the "kana" key on the keyboard inserts a space. manually changing input in the drop down from the top menu inputs the normal ascii characters.

listing folders/files with japanese just shows question marks.

The above mention problem with other UTF-8 characters like in my original report is still there

**Comment by sodiumjoe** [comment]
the utf-8 characters issue might have to do with fallback fonts not containing the necessary characters.

When I paste your input I get:

```
➤ echo 'tab:\→\ ,trail:▫<fe0e>,nbsp:•,eol:¶,extends:»,precedes:«'
tab:\→\ ,trail:▫︎,nbsp:•,eol:¶,extends:»,precedes:«
```

![2018-04-05 at 12 40 am](https://user-images.githubusercontent.com/845857/38352986-ea8ef154-3869-11e8-81b5-a62abd61fd11.png)

The japanese character issue looks like it's captured in #1101 (reported by you)

**Comment by sodiumjoe** [comment]
I think the fallback issue will be solved by #957

**Comment by jussi-kalliokoski** [comment]
Also having the same issue as OP, but everything seems to work fine in alacritty+tmux, the characters are rendered correctly and also vim doesn't choke on processing the configuration file. Maybe the issue is with some environment variables?

**Comment by jussi-kalliokoski** [comment]
Looks like setting `env.LC_CTYPE` in config to not include the locale, i.e. just `UTF-8` (same value as iTerm2) fixes the issue. Alacritty seems to use `LOCALE.ENCODING` by default, which was `en_FI.UTF-8` for me.

**Comment by sodiumjoe** [comment]
@jussi-kalliokoski are you on macos? If so, can you try [this branch](https://github.com/jwilm/alacritty/pull/1087)?

**Comment by jussi-kalliokoski** [comment]
@sodiumjoe yup, on macOS and yep, works on that branch without overrides.

**Comment by chrisduerr** [comment]
Seems like the locale issue has been fixed in #1087. Closing as resolved.
